prompts.py

```diff
@@ -1,55 +1,71 @@
-from lang import lang, all_langs
-
-proof_marker = None
-cheat_marker = None
-check_proof = None
-if lang == 'Dafny':
-    proof_marker = 'ensures'
-elif lang == 'Coq':
-    proof_marker = 'Qed'
-    cheat_marker = 'Admitted'
-elif lang == 'Lean4':
-    proof_marker = 'theorem'
-    cheat_marker = 'sorry'
-
-if check_proof is None:
-    if proof_marker:
-        check_proof = lambda v: proof_marker in v
-    else:
-        check_proof = lambda v: True
-    if cheat_marker:
-        old_check_proof = check_proof
-        check_proof = lambda v: old_check_proof(v) and cheat_marker not in v
-
-problem_fact = (f"""### Spec: In {lang}, write a factorial function and prove that the factorial is always strictly positive.
+from lang_config import LANG, ALL_LANGS
+
+if LANG == "Dafny":
+    proof_marker = "ensures"
+    cheat_marker = None
+elif LANG == "Coq":
+    proof_marker = "Qed"
+    cheat_marker = "Admitted"
+elif LANG == "Lean4":
+    proof_marker = "theorem"
+    cheat_marker = "sorry"
+
+if proof_marker:
+    CHECK_PROOF = lambda v: proof_marker in v
+else:
+    CHECK_PROOF = lambda v: True
+
+if cheat_marker:
+    CHECK_PROOF = lambda v: CHECK_PROOF(v) and cheat_marker not in v
```

Comment:
I think that will result in an infinite loop?

Reply:
thanks for catching fixed